src/classes/COFF_Affiliation_TDTM.cls

```diff
@@ -283,16 +281,25 @@ public class COFF_Affiliation_TDTM extends TDTM_Runnable {
     * member should not be at the same time teaching and attending a course, so querying for all outdatedFacultyIDs
     * should be ok.
     ********************************************************************************************************/
-    private void updateEnrollments(List<ID> outdatedFacultyIDs, List<ID> updatedFacultyIDs, DmlWrapper dmlWrapper) {
-        List<Course_Enrollment__c> oldEnrolls = [select Affiliation__c, Account__c, Course_Offering__c
-                                                 from Course_Enrollment__c where Contact__c in :outdatedFacultyIDs];
-        UTIL_Debug.debug('****Number of old enrollments: ' + oldEnrolls.size());
+    private void updateEnrollments(List<ID> outdatedFacultyIDs, List<ID> updatedFacultyIDs, Set<Id> offsWithChangedFaculty, DmlWrapper dmlWrapper) {      
+        List<Course_Enrollment__c> oldEnrolls = new List<Course_Enrollment__c>();
         List<Course_Enrollment__c> newEnrolls = new List<Course_Enrollment__c>();
-        for(Integer i = 0; i < oldEnrolls.size(); i++) {
-            newEnrolls.add(new Course_Enrollment__c(Contact__c = updatedFacultyIDs[i], Account__c = oldEnrolls[i].Account__c,
-                                                    Course_Offering__c = oldEnrolls[i].Course_Offering__c,
-                                                    Affiliation__c = oldEnrolls[i].Affiliation__c));
+        List<Course_Offering__c> updatedOfferings = [SELECT Id, Faculty__c,
```

Comment:
I think it is necessary to have the 2nd query because:
1. We are filtering the subquery with some sets of ids which are built in the loop of the 1st query. 
2. It might be a tradeoff between using another query or passing list of sobject as a paremeter.

Let me know what you think :)